canvas_modules/common-canvas/src/common-properties/actions/button/button.jsx

```diff
@@ -27,10 +27,29 @@ class ButtonAction extends React.Component {
 	constructor(props) {
 		super(props);
 		this.state = {
+			icon: null
 		};
 		this.applyAction = this.applyAction.bind(this);
 	}
 
+	// Load icon via buttonIconHandler with a new type "actionButtonIcon" after mount
+	componentDidMount() {
+		const { controller, action } = this.props;
+		const buttonIconHandler = controller.getHandlers().buttonIconHandler;
+		if (buttonIconHandler && action) {
+			const propertyId = { name: action.data.parameter_ref };
+			buttonIconHandler({
+				type: "actionButtonIcon",
+				propertyId: propertyId,
+				buttonId: action.name,
+				data: action
+			}, (appIcon) => {
+				this.setState({ icon: appIcon });
```

Comment:
Why not do this in the constructer then we wouldn't need to set the icon in the state.  That would also remove the case where `null` is nothing is returned.